components/Banner/examples/supplemental/MDCBannerView.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+@interface MDCBannerView : UIView
+
+@property(nonatomic, copy, nonnull) NSString *text;
+
+@property(nonatomic, strong, nullable) UIImage *image;  // default is nil
+
+/**
+ The buttons representing the banner's actions.
+ */
+@property(nonatomic, copy, nonnull) NSArray<__kindof UIButton *> *buttons;
```

Comment:
We typically use a model-based approach to managing buttons and their corresponding actions on components. See the ActionSheet and Alert components for examples of this.

Reply:
After discussion, since it is a view object, the action will be likely exposed on the controller in the future. The `buttons` will be kept for now.